content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,613 @@
+using different seeds and report the mean and standard deviation of each metric over all experiments rather than the result of a single training. Let's try something else. 
+
+### Zero Shot Evaluation
+Since we have at our disposition addresses from other countries, let's see if our model has really learned a typical address sequence
+or if it has simply *memorized* all the training examples.
+
+We will test our model on three different types of dataset
+
+   - first, on addresses with the exact same structure 
+    as in our training dataset: addresses from the United States of America (US) and the United Kingdom (UK)
+   - secondly, on addresses with the exact same structure as those in our training dataset **but** 
+    written in a totally different language: addresses from Russia (RU)
+   - finally, on addresses that exhibit a different structure **and** that are written in a different language: addresses from Mexico (MX).
+
+For each test, we will use a dataset of `100,000` examples in total, and we will evaluate using the best epoch of our two models (i.e. last epoch for both of them). 
+Also, we will use the same pre-processing steps as before (i.e. data vectorization, the same pad collate function), but we will only apply
+a test phase, meaning no training step.
+
+First, let's download and vectorize all the needed datasets.
+
+```python
+download_data('./data/', "us")
+download_data('./data/', "gb")
+download_data('./data/', "ru")
+download_data('./data/', "mx")
+
+us_data = pickle.load(open("./data/us.p", "rb"))  # 100,000 examples
+gb_data = pickle.load(open("./data/gb.p", "rb"))  # 100,000 examples
+ru_data = pickle.load(open("./data/ru.p", "rb"))  # 100,000 examples
+mx_data = pickle.load(open("./data/mx.p", "rb"))  # 100,000 examples
+
+dataset_vectorizer.vectorize(us_data)
+dataset_vectorizer.vectorize(gb_data)
+dataset_vectorizer.vectorize(ru_data)
+dataset_vectorizer.vectorize(mx_data)
+```
+
+##### First Test
+
+Now let's test for the United States of America and United Kingdom.
+
+```python
+us_loader = DataLoader(us_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(us_loader)
+exp_bi_lstm.test(us_loader)
+
+gb_loader = DataLoader(gb_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(gb_loader)
+exp_bi_lstm.test(gb_loader)
+```
+
+The next table presents the results of both models for both countries. We obtain
+better results for the two countries using the bidirectional bi-LSTM (around 8% better). It's interesting to see that considering that their structures are similar to those in the training dataset (Canada), we obtain near as good results as those observed during training. Meaning that our model seems to have learned to recognize the structure of an address. Also, the
+presence of the same language as in the training dataset (i.e. English), we obtain poorer results than before. That situation
```

Comment:
Not sure what you mean with this sentence, maybe : 

Also, despite the language being the same as in the training dataset (English), we obtain poorer results.